### PR TITLE
[webkitbugspy] Allow creation of new issues

### PR DESCRIPTION
**Pull request by JonWBedard**
#### a5fae230b3faa65d47ff4d18efe5ca08e1a79e4f
<pre>
[webkitbugspy] Allow creation of new issues
<a href="https://bugs.webkit.org/show_bug.cgi?id=237123">https://bugs.webkit.org/show_bug.cgi?id=237123</a>
&lt;rdar://problem/89388181 &gt;

Reviewed by Dewei Zhu.

* metadata/trackers.json: Define project.
* Tools/Scripts/libraries/webkitbugspy/setup.py: Bump version.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/__init__.py: Ditto.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/bugzilla.py:
(Tracker.projects): Return a dictionary of projects, components and versions.
(Tracker.create): Create a new bug, prompting user for product, component and version
if needed.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/github.py:
(Tracker.labels): Return a dictionary of all labels associated with the project.
(Tracker.projects): Return an empty dictionary.
(Tracker.create): Create a new issue.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/__init__.py: Export PROJECT data.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/base.py:
(Base.__init__): Mock trackers should define a project dictionary.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/bugzilla.py:
(Bugzilla.__init__):
(Bugzilla._product_details): Derive product details from project dictionary.
(Bugzilla._create): Create a new bug.
(Bugzilla.request): Handle requests for product details and bug creation.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/data.py:
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/github.py:
(GitHub.__init__):
(GitHub._labels): Return a list of project labels.
(GitHub._create): Create a new issue.
(GitHub.request): Handle requests for labels and issue creation.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/mocks/radar.py:
(RadarClient.find_components): Mock radarclient&apos;s component search API.
(RadarClient.create_radar): Mock radarclient&apos;s radar creation API.
(Radar.Component): Added.
(Radar.exceptions): Added.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/radar.py:
(Tracker):
(Tracker.Encoder.default): Pass &apos;projects&apos; to radar instances.
(Tracker.__init__): Callers should define projects to limit component scope.
(Tracker.projects): Search radar components to provide a subset relevant to the declared projects.
(Tracker.create): Create a radar, prompting user for component and version if needed.
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tests/bugzilla_unittest.py:
(TestBugzilla.test_state_why):
(TestBugzilla):
(TestBugzilla.test_projects):
(TestBugzilla.test_create):
(TestBugzilla.test_create_prompt):
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tests/github_unittest.py:
(TestGitHub.test_state_why):
(TestGitHub):
(TestGitHub.test_labels):
(TestGitHub.test_projects):
(TestGitHub.test_create):
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tests/radar_unittest.py:
(TestRadar.test_state_why):
(TestRadar):
(TestRadar.test_projects):
(TestRadar.test_create):
(TestRadar.test_create_prompt):
* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/tracker.py:
(Tracker.from_json): Pass &apos;projects&apos; to radar instances.
(Tracker.projects): Add base-class implementation.
(Tracker.create): Ditto.

Canonical link: <a href="https://commits.webkit.org/248120@main">https://commits.webkit.org/248120@main</a>
git-svn-id: <a href="https://svn.webkit.org/repository/webkit/trunk@290889">https://svn.webkit.org/repository/webkit/trunk@290889</a> 268f45cc-cd09-0410-ab3c-d52691b4dbfc
</pre>

### CODE REVIEW

File: Tools/Scripts/libraries/webkitbugspy/webkitbugspy/bugzilla.py [hunk 1]
```diff
@@ -284,3 +284,87 @@ def add_comment(self, issue, text):
...
+    @property
+    @webkitcorepy.decorators.Memoize()
+    def projects(self):
+        response = requests.get('{}/rest/product_enterable{}'.format(self.url, self._login_arguments(required=False)))
+        if response.status_code // 100 != 2:
+            sys.stderr.write("Failed to retrieve project list'\n")
+            return dict()
+
+        result = dict()
+        for id in response.json()['ids']:
```
Comment (dewei-zhu) [inline Tools/Scripts/libraries/webkitbugspy/webkitbugspy/bugzilla.py]: I would assume we always expect those keys?

File: Tools/Scripts/libraries/webkitbugspy/webkitbugspy/github.py [hunk 2]
```diff
@@ -348,3 +348,56 @@ def add_comment(self, issue, text):
...
+                continue
+            result[label['name']] = dict(
+                color=label.get('color'),
+                description=label.get('description'),
+            )
+        return result
+
+    @property
+    def projects(self):
+        # FIXME: Should be implemented via tags in GitHub, come up with a standard techinque for parsing
```
Comment (dewei-zhu) [inline Tools/Scripts/libraries/webkitbugspy/webkitbugspy/github.py]: typo: technique?

**Review by dewei-zhu — APPROVED** [review]
r=me with comment

**Comment by JonWBedard** [comment]
Landed https://commits.webkit.org/248120@main (a5fae230b3fa)!

**Pull request merged**